src/TSTransformer/nodes/statements/transformForOfStatement.ts

```diff
@@ -435,6 +437,50 @@ function getLoopBuilder(state: TransformState, node: ts.Node, type: ts.Type): Lo
 	}
 }
 
+export function transformForOfRangeMacro(state: TransformState, node: ts.ForOfStatement): luau.List<luau.Statement> {
+	// we proved this before
+	assert(ts.isCallExpression(node.expression));
+
+	if (!ts.isVariableDeclarationList(node.initializer)) {
+		DiagnosticService.addDiagnostic(errors.noRangeMacroExpressionInitializer(node.initializer));
+		return luau.list.make();
+	}
+
+	// this should be impossible to fail
+	assert(node.initializer.declarations.length === 1);
+
+	const decName = node.initializer.declarations[0].name;
+	if (!ts.isIdentifier(decName)) {
+		DiagnosticService.addDiagnostic(errors.noRangeMacroNonIdentifierInitializer(decName));
+		return luau.list.make();
+	}
```

Comment:
Could also be checked for above diagnostic, seems unnecessary to have it entirely separate?